docs/content/en/docs/features/k8s-filtering.md

```diff
@@ -0,0 +1,167 @@
+---
+title: "K8s Namespace and pod-label filtering"
+weight: 1
+description: "Tetragon in-kernel filtering using Kubernetes namespaces and pod label filters"
+---
+
+## Motivation
+
+Tetragon is configured via [TracingPolicies]({{< ref "docs/reference/tracing-policy" >}}). Broadly speaking,
+TracingPolicies define _what_ situations Tetragon should react to and _how_. The
+_what_ can be, for example, specific system calls with specific argument
+values. The _how_ defines what action the Tetragon agent should perform when
+the specified situation occurs.
+The most common action is generating and event, but there
+are others (e.g., killing the corresponding processes).
+
+Here, we are concerned with applying tracing policies only on a subset of pods
+running on the system based on their namespace, and, in future work, their
+labels.
+
+To this end, a new type of policy is introduced: `TracingPolicyNamespaced` that
+is exactly the same as the existing `TracingPolicy`, but it is _only_ applied
+to pods of the namespace that the policy is defined.
+
+As is the case with TracingPolicies, TracingPoliciesNamespaced are
+implemented in-kernel with eBPF. This is important for both observability
+and enforcement use-cases. For observability, copying only the relevant events
+from kernel- to user-space reduces overhead. For enforcement,
+performing the enforcement action in-kernel avoids the
+race-condition of doing it in user-space. For example, let us consider the case
+where we want to block an application from performing a system call. Performing
+the filtering in-kernel means that the application will never finish executing
+the system call, which is not possible if enforcement happens in user-space.
+
+To ensure that namespaced tracing policies are always correctly applied,
+Tetragon needs to perform actions before containers start executing. Tetragon
+supports this via [OCI runtime
+hooks](https://github.com/opencontainers/runtime-spec/blob/main/config.md#posix-platform-hooks).
+If such hooks are not added, Tetragon will apply policies in a best-effort
+manner using information from the k8s API server.
+
+
+## Demo
+
+For this demo, we use containerd and configure appropriate run-time hooks using minikube.
+
+First, let us start minikube, build and load images, and install Tetragon and OCI hooks:
+```
+$ minikube start --container-runtime=containerd
+$ ./contrib/rthooks/minikube-containerd-install-hook.sh
+$ make image image-operator
+$ minikube image load --daemon=true cilium/tetragon:latest cilium/tetragon-operator:latest
+$ minikube ssh -- sudo mount bpffs -t bpf /sys/fs/bpf
+$ helm install --namespace kube-system \
+	--set tetragonOperator.image.override=cilium/tetragon-operator:latest \
+	--set tetragon.image.override=cilium/tetragon:latest  \
+	--set tetragon.enablePolicyFilter="true" \
+	--set tetragon.grpc.address="unix:///var/run/cilium/tetragon/tetragon.sock" \
+	tetragon ./install/kubernetes
+```
```

Comment:
Thank you so much for documenting this! :) We will need to improve the TracingPolicy reference as well.

For this code block and the following, can you remove the `$` and `#` at the beginning of the line and add `shell` just after the three backticks, like:
````
```shell
minikube start ...
```
````
It modifies the aspect like that (top is not done correctly, down is):
<img width="694" alt="image" src="https://user-images.githubusercontent.com/11256051/227931541-052c48dd-db63-41ca-8b47-01275b092e80.png">

And not having the prompt make it easier to copy and paste.

Can you also try to separate input from the output (not using the `shell` thing with the output), adding "The output is similar to this" or other typical sentences between the two. See https://kubernetes.io/docs/contribute/style/style-guide/#separate-commands-from-output

Reply:
Thanks for the comment.

I prefer to use the `$` to indicate the shell prompt so that I can keep the output into the same box and differentiate between the two, but we need to be consistent which means that we need a common style and the kubernetes guide is as good as any (I guess). So I modified the commands to not have the shell prompt.

But, for the python test, I cannot figure out a better way to do it so I kept the python prompt. Technically, it is not "command output" so one could argue that it is not incompatible with the style guide.

Reply:
Just merged it, thanks a lot! :)